File: frontend/lib/models/location.dart
class Location {
  Location(
      {required this.id,
      required this.name,
      required this.latitude,
      required this.longitude});

  final int id;
  final String name;
  final double latitude;
  final double longitude;

  factory Location.fromJson(Map<String, dynamic> json) {
    return Location(
      id: json['id'],
      name: json['name'],
      latitude: json['latitude'].toDouble(),
      longitude: json['longitude'].toDouble(),
    );
  }
}

File: backend/src/models/location.js
const { Sequelize, DataTypes } = require("sequelize");
const sequelize = require("../db/sequelize");

const Location = sequelize.define(
  "Location",
  {
    id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      autoIncrement: true,
      primaryKey: true,
    },
    name: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    longitude: {
      type: DataTypes.DOUBLE,
      allowNull: false,
    },
    latitude: {
      type: DataTypes.DOUBLE,
      allowNull: false,
    },
  },
  {
    freezeTableName: true,
  }
);

module.exports = Location;
